### PR TITLE
PT-3759: The pedigree editor still allows duplicate external IDs

### PR DESCRIPTION
**Pull request by allasm**
@sashaandjic please check error messages, they should (probably) be reworded

### CODE REVIEW

**Comment by marta-** [comment]
@allasm Unless I am misunderstanding your concern, PT-3742 should be sufficient.

**Comment by sashaandjic** [comment]
Another problem: 
- create a new patient
- open pedigree editor; pick template with many nodes
- type existing ExternalID on more than one node that has no patient assigned 
- for each node click on "Create new"; the new patient record will be created and an error "This identifier already exists" will be presented
- move on to another node with the same ID and repeat the process
- this way you can create many patient records with identical external ID; pedigree editor will save with no error
screen shot 2018-07-23 at 1 52 51 pm

screen shot 2018-07-23 at 1 53 55 pm

screen shot 2018-07-23 at 1 54 40 pm

**Comment by allasm** [comment]
@sashaandjic all comments/bugs have been addressed

**Comment by sashaandjic** [comment]
i'm still able to create multiple records with same identifier within Pedigree editor.
- create patient
- click on pedigree thumbnail and select child/mother/father template
- click on any parent node; type "abc" into identifier field and click on "Create new" bttn
- click on other parent node ; type "abc" into identifier field and click on "Create new" button
- click on Save and exit the pedigree
- there are two patient records created with the same ID

File: components/pedigree/resources/src/main/resources/pedigree/externalIdManager.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        },
+
+        duplicateIDPresent: function()
+        {
+            for(var id in this.externalIDs) {
+                if (this.externalIDs.hasOwnProperty(id)) {
+                    if (this.externalIDs[id].length > 1) {
+                        var involvesExternalPatient = false;
+                        if (Helpers.arrayContains(this.externalIDs[id], this._EXTERNAL_PATIENT)) {
+                            involvesExternalPatient = true;
```
Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/externalIdManager.js]: `var involvesExternalPatient = Helpers.arrayContains(this.externalIDs[id], this._EXTERNAL_PATIENT);`

**Pull request merged**